modules/navier_stokes/src/fvkernels/WCNSFV2PMomentumDriftFlux.C

```diff
@@ -26,6 +26,7 @@ WCNSFV2PMomentumDriftFlux::validParams()
   params.addParam<MooseFunctorName>("w_slip", "The slip velocity in the z direction.");
   params.addRequiredParam<MooseFunctorName>("rho_d", "Dispersed phase density.");
   params.addParam<MooseFunctorName>("fd", 0.0, "Fraction dispersed phase.");
+  params.set<unsigned short>("ghost_layers") = 5;
```

Comment:
Could you try if this works if you use mesh refinement and more processors for the run? If you only use 2 processors then for `nx=10` (current setting) with a 5 layer ghosting every process has access to every element. Also, could you try printing the matrix? I have a feeling that it will be full due to super long chains between gradient calls. If we hit that we don't have another option, just using the old value.